test/metabase/driver/query_processor_test.clj

```diff
@@ -481,12 +488,28 @@
 ;; # POST PROCESSING TESTS
 
 ;; ## LIMIT-MAX-RESULT-ROWS
-;; Apply limit-max-result-rows to an infinite sequence and make sure it gets capped at `max-result-rows`
-(expect max-result-rows
+;; Apply limit-max-result-rows to an infinite sequence and make sure it gets capped at `absolute-max-results`
+(expect absolute-max-results
   (->> (((u/runtime-resolved-fn 'metabase.driver.query-processor 'limit) identity) {:rows (repeat [:ok])})
        :rows
        count))
 
+;; Apply an arbitrary max-results on the query and ensure our results size is appropriately constrained
+(expect 1234
+  (->> (((u/runtime-resolved-fn 'metabase.driver.query-processor 'limit) identity) {:constraints {:max-results 1234}
```

Comment:
there's not real benefit to using `u/runtime-resolved-fn` here. That's just a way to resolve a fn for repeated use without the overhead of calling `resolve` every time. Just do 

```clojure
@(resolve 'metabase.driver.query-processor 'limit)
```

instead